commands/ci/artifact/artifact.go

```diff
@@ -0,0 +1,65 @@
+package ci
+
+import (
+	"archive/zip"
+	"io"
+	"os"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/profclems/glab/api"
+	"github.com/profclems/glab/commands/cmdutils"
+	"github.com/spf13/cobra"
+	"github.com/xanzy/go-gitlab"
+)
+
+func NewCmdRun(f *cmdutils.Factory) *cobra.Command {
+	var jobArtifactCmd = &cobra.Command{
+		Use:     "artifact <refName> <jobName> [flags]",
+		Short:   `Download all Artifacts from the last pipeline`,
+		Aliases: []string{"push"},
+		Example: heredoc.Doc(`
+	$ glab ci artifact main build
+	$ glab ci artifact main deploy --path="artifacts/"
+	`),
+		Long: ``,
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			repo, err := f.BaseRepo()
+			if err != nil {
+				return err
+			}
+			apiClient, err := f.HttpClient()
+			if err != nil {
+				return err
+			}
+			p, err := cmd.Flags().GetString("path")
+			if err != nil {
+				return err
+			}
+
+			artifact, _ := api.DownloadArtifactJob(apiClient, repo.FullName(), args[0], &gitlab.DownloadArtifactsFileOptions{Job: &args[1]})
+
+			zr, _ := zip.NewReader(artifact, artifact.Size())
+			os.Mkdir(p, 0777)
+			for _, v := range zr.File {
+				if v.FileInfo().IsDir() {
+					os.Mkdir(p+v.Name, 0777)
```

Comment:
The error from `os.Mkdir` shouldn't be ignored. You need to check it and return it if something goes wrong.

Also, the rights of 777 are not correct here. I think something like 644 is more suitable.